src/core/datastore/processors/offline-data-processor.js

```diff
@@ -105,10 +108,14 @@ export class OfflineDataProcessor extends DataProcessor {
   }
 
   _addMetadataToEntities(data) {
-    ensureArray(data).forEach((entity) => {
+    const siSingle = !Array.isArray(data);
```

Comment:
Assuming this is checking whether data is a single entity or an array - wouldn't you want to add the metadata in case it is a single entity?

Reply:
On a second look, maybe your `ensureArray()` converts a single entity into an array of 1.
A little confusing, nevertheless.

Reply:
That is what `ensureArray()` does, yes. Which part is confusing? You ensure something is an array, then you call a `map()` on that array, deep clone each entity and add metadata to it. In the end you return the result in the expected form (whatever was passed), one entity or an array of entities.

Reply:
I think the part that always confuses me is the use of the word `ensure`. Does `ensureArray(data)` imply we're validating that `data` is an array, or does it mean we'll convert `data` to an array if it's not an array? 😕

Reply:
We're making sure it's an array :) Meaning we'll convert it. Looking at the code would clarify in a second, but reading it as text does allow for some ambiguity. `putInArrayIfNotAnArray()`? :) I'm open to suggestions.